CRM/Contribute/Form/AdditionalInfo.php

```diff
@@ -271,11 +271,11 @@ public static function postProcessCommon(&$params, &$formatted, &$form) {
       $formatted[$f] = $params[$f] ?? NULL;
     }
 
-    if (!empty($params['thankyou_date']) && !CRM_Utils_System::isNull($params['thankyou_date'])) {
-      $formatted['thankyou_date'] = CRM_Utils_Date::processDate($params['thankyou_date'], $params['thankyou_date_time']);
+    if (!empty($params['thankyou_date'])) {
+      $formatted['thankyou_date'] = CRM_Utils_Date::processDate($params['thankyou_date']);
```

Comment:
I don't think we need to do the processDate any more either - if you remove it & the test still passes it isn't needed

Reply:
The problem if I do that is that if you then just enter a time on the form with no date it crashes: `Incorrect datetime value: '4' for column thankyou_date`, where "4" is whatever the hour value is. It's an odd thing for someone to enter in this particular situation, but it lets you do it. So either processDate or a formRule or ?

Reply:
@demeritcowboy OK - we can leave that for another day then - this is definitely an improvment as is